### DISCUSSION TITLE
how to set a custom template for all pages inside a certain directory?

### DISCUSSION THREAD

**Discussion by plgruener** [getpelican/pelican#3270] [Q&A] [answered]
I'm writing a custom theme for a non-blog site, and I want all pages in subdirectory A use one custom template (say `theme/templates/page_a.html`), all pages in subdir B another template (`theme/templates/page_b.html`) and all other pages use the default/generic page template.

I know I can override the `template:…` metadata per file, but this is tedius and error-prone.  
I could also write a custom plugin, but I'd rather not if there is an easy builtin way. Any ideas?

**Comment by avaris** [comment, marked as answer]
See [`EXTRA_PATH_METADATA`](https://docs.getpelican.com/en/latest/settings.html#metadata):

```python
EXTRA_PATH_METADATA = {
    'A': {'template': 'page_a'},
    'B': {'template': 'page_b'},
    # ...
}

**Reply by avaris** [reply]
PS: Just make sure folder names are distinct enough. Because `'A'` will capture both `A/*` and `A_file.md` etc. It's essentially a prefix check on the path.

**Reply by plgruener** [reply]
thank you so much, worked perfectly. I missed that this is a prefix check (not very intuitiv for a dictionary key).
